mesonbuild/msubprojects.py

```diff
@@ -435,6 +436,33 @@ def update(self) -> bool:
             self.wrap.update_hash_cache(self.repo_dir)
         return success
 
+    def syncwrap(self) -> bool:
+        if self.wrap.type != 'git' and not self.wrap.redirected:
+            return True
+
+        self.log(f'Sync {self.wrap.name}...')
+        if self.wrap.redirected:
+            self.wrap.filename = Path(self.wrap.original_filename).resolve()
+            self.wrap.parse_wrap()
```

Comment:
I'm not sure we should update redirected wraps, they are usually not in the same git repo as the main project.

Reply:
There's a warning when this command updates a redirected wrap. I think it is better to warn the user that there are changes in a redirect than silently ignore them.

Reply:
Reading this again, if I understand correctly you override the original [wrap-redirect] with the [wrap-git]? That actually sounds useful and maybe we should not limit that to wrap-git. Replacing redirects with wrap-file seems useful too so main project can commit them too.

Reply:
No, it modifies the original [wrap-git] and warns the user it did. It leaves the [wrap-redirect] untouched. Your idea is interesting but i think it would belong to another subcommand : the decision to promote a redirect to the current project is related to the source code and not just its build system.
So in the present case, to have wraps that matches its directories, the user will need to fix the warning by commiting in the subproject which owns the redirect and re-run syncwrap to get the new subproject commit.

Reply:
Why are you doing `self.wrap.filename = Path(self.wrap.original_filename).resolve()` then?

Reply:
You are right, no need to resolve here.
To summarize, given the subprojects A depending on B, if there's a commit in B, subprojects/A/subprojects/B.wrap will be updated and a warning issued to the user.